### ISSUE TITLE
PostgreSQL LISTEN/NOTIFY instead of polling?

### ISSUE THREAD

**Issue by justinclift** [rverton/pgjobs#2]
As a general thought while looking over the Readme, the polling aspect of things seems a bit non-optimal.

Kind of wondering if using PostgreSQL's `LISTEN` / `NOTIFY` would be a useful way to improve upon that, at least for jobs that aren't queued for the future:

* https://www.postgresql.org/docs/current/sql-listen.html
* https://www.postgresql.org/docs/current/sql-notify.html

**Comment by rverton** [comment]
Yeah, using a listen/notify approach is indeed better, but it will make this more complex. If I find some time I will see if I can do this in a simple way. I will leave this open for now.
